keps/sig-network/2732-np-version/README.md

```diff
@@ -0,0 +1,803 @@
+
+	// Kubernetes 1.8 brought the NetworkPolicy API to v1, adding egress policies
+	// and ipBlocks.
+	NetworkPolicyVersion1_8 NetworkPolicyVersion = "1.8"
+
+	// Kubernetes 1.9 added alpha IPv6 support. Plugins implementing this version are
+	// assumed to recognize IPv6 addresses, even if they don't implement them.
+	NetworkPolicyVersion1_9 NetworkPolicyVersion = "1.9"
+
+	// Kubernetes 1.11 added support for specifying podSelector and namespaceSelector
+	// together in a peer.
+	NetworkPolicyVersion1_11 NetworkPolicyVersion = "1.11"
+
+	// Kubernetes 1.12 added alpha SCTP support. Plugins implementing this version are
+	// assumed to recognize `protocol: SCTP`, even if they don't implement it.
+	NetworkPolicyVersion1_12 NetworkPolicyVersion = "1.12"
+)
+
+```
+
+And add a new `Status NetworkPolicyStatus` field to `NetworkPolicy`, defined as:
+
+```
+// NetworkPolicyStatus contains information about the processing of a NetworkPolicy
+type NetworkPolicyStatus struct {
+	// conditions associated with the policy. These may include a "Supported"
+	// condition indicating whether the features used by the policy are supported,
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []NetworkPolicyStatusCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,3,rep,name=conditions"`
+}
+
+// NetworkPolicyStatusConditionType is a valid value of NetworkPolicyStatusCondition.Type
+type NetworkPolicyStatusConditionType string
+
+const (
+	// NetworkPolicyStatusSupported - The plugin supports this NetworkPolicy.
+	NetworkPolicyStatusSupported NetworkPolicyStatusConditionType = "Supported"
+
+	// NetworkPolicyStatusProblem - The NetworkPolicy has a problem
+	NetworkPolicyStatusProblem NetworkPolicyStatusConditionType = "Problem"
+)
+
+// NetworkPolicyStatusCondition contains details about the state of NetworkPolicy processing
+type NetworkPolicyStatusCondition struct {
+	Type   NetworkPolicyStatusConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=NetworkPolicyStatusConditionType"`
+	Status ConditionStatus                  `json:"status" protobuf:"bytes,2,opt,name=status,casttype=ConditionStatus"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+	// Short, machine understandable string that gives the reason for condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty" protobuf:"bytes,5,opt,name=reason"`
+	// Human-readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message,omitempty" protobuf:"bytes,6,opt,name=message"`
+}
+```
+
+### Semantics
+
+When a user creates a NetworkPolicy, and is using a network plugin that
+implements this specification:
+
+  - If the NetworkPolicy uses API fields which are not known to the
```

Comment:
In general this isn't true.  kubectl does some validation but that can be bypassed and the apiserver drops unknown fields

Reply:
yeah, this was the big problem that led to the original KEP being abandoned